drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/rulesengine/AbstractRulesEvaluator.java

```diff
@@ -168,9 +170,12 @@ protected List<Match> process(Map<String, Object> factMap, boolean processEventI
         List<Match> matchList = atomicRuleEvaluation(processEventInsertion,
                                                      () -> insertFacts(factMap, processEventInsertion),
                                                      (fhs, matches) -> {
-                                                         if (log.isDebugEnabled()) {
-                                                             for (InternalFactHandle fh : fhs) {
-                                                                 if (fh.isDisconnected()) {
+                                                         for (InternalFactHandle fh : fhs) {
+                                                             if (fh.isDisconnected()) {
+                                                                 if (fh instanceof DefaultEventHandle eventHandle) {
+                                                                     eventHandle.unscheduleAllJobs((ReteEvaluator) rulesExecutorSession.asKieSession());
```

Comment:
I think a problem is in `PropagationEntry.Insert.propagate`:
https://github.com/apache/incubator-kie-drools/blob/main/drools-core/src/main/java/org/drools/core/phreak/PropagationEntry.java#L239-L242

When an inserted event doesn't match, handle is disconnected and removed from objectStore. I think `unscheduleAllJobs` should be also called in the method. But this PR is a workaround that we can avoid updating drools in drools-ansible-rulebook-integration.